File: El Mamoun/Lv 1/Problem 30.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

/*
 Problem 30: Write a program to calculate factorial of N!
 */
namespace Problem_30
{
    internal class Program
    {

      static int ReadPostiveNumber(string Message)
        {
           
            int Number = 0;
            do
            {
                Console.WriteLine(Message);
                Number = Convert.ToInt32(Console.ReadLine());

            } while (Number < 0);

            return Number;
        }   

      static int Factorial(int Number)
        {
            int Fact = 1;

            for(int Counter =Number; Counter >= 1; Counter--)
            {
                Fact *= Counter;
            }

            return Fact;
        }

      
        static void Main(string[] args)
        {

            Console.WriteLine("Facorial Number is: "+Factorial(ReadPostiveNumber("Please Enter a Number?: ")));


            Console.ReadKey();
        }
    }
}

File: El Mamoun/Lv 1/Problem 30.cpp
#include<iostream>
using namespace std;

/*
 Problem 30: Write a program to calculate factorial of N!
 */

int ReadPostiveNumber(string Message)
{
	int Number = 0;

	do 
	{
		cout << Message;
		cin >> Number;

	} while (Number < 0);

	return Number;
}

int Factorial(int Number)
{
	int Factorial = 1;
	
	for (int Counter = Number; Counter >= 1; Counter--)
	{
		Factorial *= Counter;

	}
	return Factorial;
}



int main()
{

	cout << Factorial(ReadPostiveNumber("Please Enter a Postive Number? "));

	return 0;
}